test/translator/common_test.cc

```diff
@@ -151,35 +151,25 @@ TEST(Common, ShouldFailBuildAllocationWhenAllocPageFails) {
 }
 
 TEST(Common, SafePointerCastWrite) {
-  uint32_t expected_val = 0;
-  if (translator::IsLittleEndian())
-    expected_val = 0x13292022;
-  else
-    expected_val = 0x22202913;
-  uint8_t buf[100];
-
-  uint32_t* val = translator::SafePointerCastWrite<uint32_t>(buf);
+  uint32_t expected_val = 0x13292022;
+
+  uint32_t write_buffer = 0;
+  uint8_t* buf = reinterpret_cast<uint8_t*>(&write_buffer);
+  translator::Span<uint8_t> span(buf, sizeof(uint32_t));
+
+  uint32_t* val = translator::SafePointerCastWrite<uint32_t>(span);
   *val = expected_val;
```

Comment:
Both tests should check that the returned pointer isn't `nullptr` before dereferencing it.